src/libsyntax/parse/parser.rs

```diff
@@ -4330,6 +4330,26 @@ impl<'a> Parser<'a> {
 
     fn parse_fn_args(&mut self, named_args: bool, allow_variadic: bool)
                      -> PResult<'a, (Vec<Arg> , bool)> {
+
+        /// Create a placeholder argument
+        fn dummy_arg(span: Span) -> Arg {
```

Comment:
Nit: error reporting is a "cold" code for a reader trying to understand the main logic (syntax of fn arguments in this case), it often makes sense to factor it into a separate function (already done) and move it somewhere else, so it's invisible unless intentionally looked for.